services/venice-controller/src/main/java/com/linkedin/venice/ingestion/control/RealTimeTopicSwitcher.java

```diff
@@ -165,6 +165,16 @@ void ensurePreconditions(
           false, // Note: do not enable RT compaction! Might make jobs in Online/Offline model stuck
           minISR,
           false);
+      if (version.isSeparateRealTimeTopicEnabled()) {
+        getTopicManager().createTopic(
+            pubSubTopicRepository.getTopic(Version.composeSeparateRealTimeTopic(store.getName())),
+            partitionCount,
+            replicationFactor,
+            StoreUtils.getExpectedRetentionTimeInMs(store, hybridStoreConfig.get()),
+            false, // Note: do not enable RT compaction! Might make jobs in Online/Offline model stuck
```

Comment:
I am wondering the case if the existing store is upgraded from non-A/A hybrid -> A/A and then enable this config or it is default enabled by cluster level config, but since RT topic has been created before, then this path will be ignored? Do you think it can be move out of this if else block and check separately?

Reply:
This is a valid concern. I can add a separate check for this config.